tests/functional/heat_controller_test.go

```diff
@@ -188,19 +290,107 @@ var _ = Describe("Heat controller", func() {
 			//keystone := GetKeystoneAPI(keystoneAPI)
 			Expect(th.GetConfigMap(configataCM).Data["heat.conf"]).Should(
 				ContainSubstring("stack_domain_admin = heat_stack_domain_admin"))
+		})
+	})
 
+	When("DB is created", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateHeat(heatName, GetDefaultHeatSpec()))
+			DeferCleanup(
+				k8sClient.Delete, ctx, CreateHeatSecret(namespace, SecretName))
+			rmqSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rabbitmq-secret",
+					Namespace: namespace,
+				},
+			}
```

Comment:
We do this quite a bit now. Maybe it's worth adding a function to `base_test.go` to save us reconstructing the `corev1.Secret{}` each and every time.

But it's a cosmetic thing, we can look at it later.